docker/src/main/java/org/jclouds/docker/domain/Network.java

```diff
@@ -101,7 +101,30 @@ public static Details create(String endpoint, String macAddress, String ipv4addr
    @SerializedNames({ "Name", "Id", "Scope", "Driver", "IPAM", "Containers", "Options" })
    public static Network create(String name, String id, String scope, String driver, IPAM ipam,
                                 Map<String, Details> containers, Map<String, String> options) {
-      return new AutoValue_Network(name, id, scope, driver, ipam, copyOf(containers), copyOf(options));
+      return builder().name(name).id(id).scope(scope).driver(driver).ipam(ipam).containers(copyOf(containers)).options(copyOf(options)).build();
    }
 
+   public static Builder builder() {
+      return new AutoValue_Network.Builder();
+   }
+
+   @AutoValue.Builder
+   public abstract static class Builder {
+
+      public abstract Builder name(@Nullable String name);
+
+      public abstract Builder id(@Nullable String id);
+
+      public abstract Builder scope(@Nullable String scope);
+
+      public abstract Builder driver(@Nullable String driver);
+
+      public abstract Builder ipam(IPAM ipam);
+
+      public abstract Builder containers(Map<String, Details> containers);
+
+      public abstract Builder options(Map<String, String> options);
```

Comment:
Enforce immutable collections in auto builders. You can follow [this pattern](https://github.com/jclouds/jclouds-labs/blob/master/azurecompute-arm/src/main/java/org/jclouds/azurecompute/arm/domain/Subnet.java#L67-L84). Basically it consists of providing a hidden builder method that will delegate to auto-value, and have the public one enforcing the immutable collections.